src/Containers/StoreHoursList/StoreHoursList.tsx

```diff
@@ -0,0 +1,529 @@
+    };
+    
+    const saveHours = (categoryName: string): void => {
+        if (Object.values(checkboxes).includes(true)) {
+            const timedArray = allCategoriesWithHours.find((categorySchedule: ICategoryWithHoursTypes): ICategoryWithHoursTypes | null | boolean => categorySchedule.category === categoryName);
+            if (timedArray !== undefined) {
+                Object.entries(checkboxes).map((checkbox): ICategoryWithHoursTypes | null => {
+                    if (checkbox[CHECKBOX_TIME] && timedArray.hoursByDay[checkbox[CHECKBOX_DAY]].length >= 1) { // cannot have more than one time per day
+                        setError('Only one time allowed');
+                    } else if (checkbox[CHECKBOX_TIME] && timedArray.hoursByDay[checkbox[CHECKBOX_DAY]].length === 0) { // add time
+                        const convertedTime = convertDateToHours(storeHours);
+                        if (convertedTime !== null) {
+                            timedArray.hoursByDay[checkbox[CHECKBOX_DAY]].push({
+                                from: convertedTime.from,
+                                to: convertedTime.to
+                            });
+                            setCheckboxes(initialCheckboxState);
+                            return timedArray;
+                        }
+                    }
+                    return null;
+                })
+            }
+        }
+    };
+
+    const getActiveSchedule = (categoryName: string): ICategoryWithHoursTypes => {
+        setActiveCategory(categoryName);
+        const activeSchedule = allCategoriesWithHours.find((el): ICategoryWithHoursTypes | null | boolean => categoryName === el.category);
+        if (activeSchedule) {
+            return activeSchedule;
+        }
+        return activeCategorySchedule;
+    };
+
+    useEffect((): void => {
+        setActiveCategorySchedule(getActiveSchedule(activeCategory))
+    }, [activeCategorySchedule]);
+
+    return (    
+        <StoreHoursContainer>
+            <SettingsCard heading={constants.TITLES.HEADING} icon={BusinessTime}>
+                <ButtonsContainer>
+                    <StyledButton
+                        icon={Edit}
+                        onClick={(): void => {
+                            setError('');
+                            setEditModalState(!editModalState);
+                        }}
+                    >
+                        { constants.BUTTONS.EDIT }
+                    </StyledButton>
+                    <StyledButton onClick={(): void => setIs24(!is24)}> 
+                        { constants.BUTTONS.TOGGLE } 
+                    </StyledButton>
+                </ButtonsContainer>
+                <StyledHeading type='h6'>
+                    { constants.TITLES.OPERATIONS }
+                    { activeCategorySchedule.category }
+                </StyledHeading>
+                {Object.entries(activeCategorySchedule.hoursByDay).map((day): React.ReactElement | null => {
```

Comment:
This can be a component in itself to help keep the bulk down here

Reply:
Would this category not rely on too many other states to make into a separate component? (e.g. is24 toggle on/off to make it AM/PM and 24 hours) and setting the active Schedule. Or would you put these as props?